backend/src/main/java/ai/giskard/service/ee/LicenseService.java

```diff
@@ -0,0 +1,156 @@
+        return license == null ? defaultLicense : license;
+    }
+
+    /**
+     * Checks if there currently is a license. If yes, parses and loads it.
+     */
+    @PostConstruct
+    public void loadLicense() throws IOException {
+        if (Files.exists(fileLocationService.licensePath())) {
+            String licenseFile = Files.readString(fileLocationService.licensePath());
+            decodeLicense(licenseFile);
+        }
+    }
+
+    /**
+     * Takes a license file content, parses it, validates that it is a proper license.
+     * If it's a real license, saves it in giskard home and updates available feature flags
+     *
+     * @param licenseFile
+     */
+    public void uploadLicense(String licenseFile) throws IOException {
+        decodeLicense(licenseFile);
+        Files.write(fileLocationService.licensePath(), licenseFile.getBytes());
+    }
+
+    private License decodeLicense(String lic) throws IOException {
+        // 1. Remove start/end decorators
+        String encodedPayload = lic.replaceAll("(^-----BEGIN LICENSE FILE-----\\n|\\n|-----END LICENSE FILE-----\\n$)", "");
+
+        // 2. Decode base64 => json
+        byte[] payloadBytes = Base64.getDecoder().decode(encodedPayload);
+        String payload = new String(payloadBytes);
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode payloadJson = mapper.readTree(payload);
+
+        // 3. Parse Json
+        String encodedData = payloadJson.get("enc").asText();
+        String encodedSig = payloadJson.get("sig").asText();
+        String algorithm = payloadJson.get("alg").asText();
+
+        if (!"base64+ed25519".equals(algorithm)) {
+            throw new GiskardRuntimeException("License file is invalid.");
+        }
+
+        // 4. Decode signing bytes and use signature to validate
+        byte[] publicKeyBytes = Hex.decode(SIGNATURE_KEY);
+        byte[] signatureBytes = Base64.getDecoder().decode(encodedSig);
+        byte[] encDataBytes = String.format("license/%s", encodedData).getBytes();
+
+        Ed25519PublicKeyParameters verifierParams = new Ed25519PublicKeyParameters(publicKeyBytes, 0);
+        Ed25519Signer verifier = new Ed25519Signer();
+
+        verifier.init(false, verifierParams);
+        verifier.update(encDataBytes, 0, encDataBytes.length);
+
+        if (!verifier.verifySignature(signatureBytes)) {
+            throw new GiskardRuntimeException("License file is invalid.");
+        }
+
+        // 5. Decode license and parse it into a License object
+        String decodedLicense = new String(Base64.getDecoder().decode(encodedData));
+        JsonNode licenseJson = mapper.readTree(decodedLicense);
+
+        JsonNode attributes = licenseJson.get("data").get("attributes");
+        JsonNode included = licenseJson.get("included");
+        if (!"ACTIVE".equals(attributes.get("status").asText())) {
+            throw new GiskardRuntimeException("License file is invalid.");
+        }
+
+        // TODO: Check expiration
```

Comment:
From keygen docs:

You should assert the following:

- That issued is not greater than the current time, indicating the user has set their system clock to the past, also known as clock tampering.
- That expiry is not less than the current time, indicating an expired license file.